os/mm/mm_heap/mm_getsize.c

```diff
@@ -90,3 +90,37 @@ size_t mm_get_largest_freenode_size(void)
 
 	return largest_size;
 }
+
+/****************************************************************************
+ * Name: mm_get_heap_free_size
+ *
+ * Description:
+ *   returns free size in the heap
+ ****************************************************************************/
+#ifdef CONFIG_DEBUG_MM_HEAPINFO
+size_t mm_get_heap_free_size(void)
+{
+	struct mm_heap_s *heap;
+
+#ifdef __KERNEL__
+	size_t free_size = 0;
+	int heap_idx;
+	heap = g_kmmheap;
+	struct mm_heap_s *target_heap;
+
+	/* Kernel can have multi-heap, so traversing all n-heaps and add all free sizes. */
+	for (heap_idx = HEAP_START_IDX; heap_idx <= HEAP_END_IDX; heap_idx++) {
+		target_heap = &heap[heap_idx];
+		free_size += (target_heap->mm_heapsize - target_heap->total_alloc_size);
+	}
+#else
+	/* User can only have single heap. */
+	heap = BASE_HEAP;
+	if (!heap) {
+		mdbg("Fail to find the heap.\n");
```

Comment:
Can you pls change the debug statement to give more information?
Like "Failed to retrieve the heap region for xxx application".
Also at line 86.

Reply:
Modified. Thanks.